### PR TITLE
Updated guide to MP Rest Client 1.3 (#97)

### PR DESCRIPTION
**Pull request by evelinec**
* Updated to features in mp rest client 1.3

* More updates for mpRestClient 1.3

* Minor changes

* Formatted java classes

* Addressed dev content review feedback

* Addressed SME review feedback

* Added links for CDI and RestClientBuilder

* Minor changes

### CODE REVIEW

**Comment by evelinec** [comment]
@Charlotte-Holt This PR is ready for you to review. You can access the staging site, https://qa-guides.mybluemix.net/guides/microprofile-rest-client.html for the full content. Feel free to add on to this PR for comments or open a new issue on this repo.

**Comment by t-koman** [comment]
@evelinec @Charlotte-Holt 
Here are my draft edits.   Let me know if you have questions/concerns.

(1) change --->The @Dependent and @RegisterRestClient annotations imply that the interface is manageable through CDI. You must have them in order to inject the client.
TO
The @Dependent and @RegisterRestClient annotations imply that the interface is manageable through CDI. Y**ou must have these annotations to inject the client.**

(2) change ---> The @Dependent annotation tells the service that the scope of the interface depends on the class that it is injected into.
TO
The @Dependent annotation tells the service that the scope of the interface depends on the class that **the annotation** is injected into.

(3) change ---> The @Dependent and @RegisterRestClient annotations imply that the interface is manageable through CDI. You must have them in order to inject the client.
TO
The @Dependent and @RegisterRestClient annotations imply that the interface is manageable through CDI **and can be injected into the client.**

(4)change --> The inventory service might also connect with a host other than the default localhost host.
TO
The inventory service **can** connect with a host other than the default localhost host.

(5) change ---> Then, you cannot configure a base URL that is not yet known. Therefore, set the host name as a variable and build the client by using the RestClientBuilder method.
TO
You cannot **then** configure a base URL that is not yet known. **In this case**, set the host name as a variable and build the client by using the RestClientBuilder method.

(6) change ---> When the server is running, select either approach to fetch your system properties:
TO
When the server is running, select **one of these approaches** to fetch your system properties:

(7)change ---> to bulleted list


**-  testDefaultLocalhost() fetches and compares system properties from the http://localhost:9080/inventory/systems/localhost URL.

-  testRestClientBuilder() gets your IP address. Then, use your IP address as the host name to fetch your system properties and compare them.**

(8) change --->  If a test failure occurs, you might have introduced a bug into the code.
TO
 **A test failure signals you might have introduced a bug into the code.**

(9)change ---> Feel free to try one of the related guides. They demonstrate more technologies that you can learn and expand on what you built here.
TO
**Try one of the other related guides where you can learn more technologies and expand on what you built here.**

File: README.adoc [hunk 1]
```diff
@@ -241,18 +251,18 @@ The helper function invokes the `system` service by calling the [hotspot=61 file
 The `inventory` service might also connect with a host other than the default `localhost` host. Then, you cannot configure a base URL that is not yet known.
```
Comment (t-koman) [inline README.adoc]: Change
The `inventory` service might also connect with a host other than the default `localhost` host. Then, you cannot configure
TO
The `inventory` service **can** connect with a host other than the default `localhost` host. **You cannot then**

File: README.adoc [hunk 2]
```diff
@@ -316,13 +326,16 @@ Results :
...
 Rerun the Maven build. You see a test failure occur.
 
 == Great work! You're done!
 
 You just invoked a remote service by using a template interface with MicroProfile Rest Client in Open Liberty.
 
+MicroProfile Rest Client also provides a uniform way to configure SSL for the client.
+You can learn more in the https://openliberty.io/blog/2019/06/21/microprofile-rest-client-19006.html#ssl[Hostname verification with SSL on Open Liberty and MicroProfile Rest Client^] blog and the https://github.com/eclipse/microprofile-rest-client/releases[MicroProfile Rest Client specification^].
+
 Feel free to try one of the related guides. They demonstrate more technologies that you can learn and
```
Comment (t-koman) [inline README.adoc]: change
Feel free to try one of the related guides. They demonstrate more technologies that you can learn and
expand on what you built here.
TO
**Try one of the other related guides where you can learn more technologies and expand on what you built here.**

File: README.adoc [hunk 3]
```diff
@@ -178,60 +185,63 @@ The [hotspot=15-18 file=2]`handles()` method inspects the HTTP response code to
...
 
-The [hotspot=13 file=1]`@Dependent` and [hotspot=14 file=1]`@RegisterRestClient` annotations imply that the interface is manageable through CDI. You must have them in order to inject the client.
+The [hotspot=RegisterRestClient file=1]`baseUri` value can also be set in the [hotspot=RegisterRestClient file=1]`@RegisterRestClient` annotation. However, this value will be overridden by the base URI property defined in the [hotspot file=0]`microprofile-config.properties` configuration file. The property defined in the MicroProfile Config file takes precedence. In a production environment, you can use the `baseUri` variable to specify a different URI for development and testing purposes. For this guide, try to specify an invalid URI value for the [hotspot=RegisterRestClient file=1]`baseUri` variable and comment out the [hotspot=baseUri file=0]`mp-rest/uri` property in the [hotspot file=0]`microprofile-config.properties` file. Then, run the `mvn install` command to verify that the `InventoryEndpointTest` fails to connect to the `system` service.
```
Comment (t-koman) [inline README.adoc]: change
However, this value will be overridden by the base URI property defined in the [hotspot file=0]`microprofile-config.properties` configuration file. The property defined in the MicroProfile Config file takes precedence.
TO
However, this value will be overridden by the base URI property defined in the [hotspot file=0]`microprofile-config.properties` configuration file, **which takes precedence.**

File: README.adoc [hunk 4]
```diff
@@ -178,60 +185,63 @@ The [hotspot=15-18 file=2]`handles()` method inspects the HTTP response code to
...
-By default, interfaces have a `@Dependent` scope unless another scope is defined on the interface.
+The [hotspot=RegisterRestClient file=1]`@RegisterRestClient` annotation registers the interface as a RESTful client. The runtime creates a CDI managed bean for every interface that is annotated with the `@RegisterRestClient` annotation.
 
-The [hotspot=14 file=1]`@RegisterRestClient` annotation registers the interface as a RESTful client. The runtime creates a CDI managed bean for every interface that is annotated with the `@RegisterRestClient` annotation.
+The [hotspot=RegisterRestClient file=1]`configKey` value in the [hotspot=RegisterRestClient file=1]`@RegisterRestClient` annotation replaces the fully-qualified classname of the properties in the [hotspot file=0]`microprofile-config.properties` configuration file.
+For example, the `<fully-qualified classname>/mp-rest/uri` property becomes `systemClient/mp-rest/uri`.
+The benefit of using Config Keys is when multiple client interfaces have the same `configKey` value, the interfaces can be configured with a single MP config property.
```
Comment (t-koman) [inline README.adoc]: change
The benefit of using Config Keys is when multiple client interfaces have the same `configKey` value, the interfaces can be configured with a single MP config property.
TO
**The benefit of using Config Keys the interfaces to be configured with a single MP config property when multiple client interfaces have the same `configKey` value,**

File: README.adoc [hunk 5]
```diff
@@ -136,39 +143,39 @@ In the [hotspot=17-22 file=2]`SystemClient` interface, add a response exception
 
 Error handling is an important step to ensure that the application can fail safely. If there is an error response such as `404 NOT FOUND` when invoking the remote service, you need to handle it. First, define an exception, and map the exception with the error response code. Then, register the exception mapper in the client interface.
```
Comment (t-koman) [inline README.adoc]: change
you need to handle it. First, define an exception, and map the exception with the error response code. Then, register the exception mapper in the client interface.
TO
**you need to define an exception and map the it with the error response code. Then, register the exception mapper in the client interface.**

File: README.adoc [hunk 6]
```diff
@@ -111,23 +113,28 @@ The code for the `system` service in the `src/main/java/io/openliberty/guides/sy
...
 
-When the [hotspot=21 file=2]`getProperties()` method is invoked, the [hotspot=17-22 file=2]`SystemClient` instance sends a GET request to the `<baseUrl>/properties` endpoint, where `<baseUrl>` is the default base URL of the `system` service. You will see how to configure the base URL in the next section.
+The [hotspot=Produces file=2]`@Produces` annotation specifies the media (MIME) type of the expected response. The default value is `MediaType.APPLICATION_JSON`.
 
-The [hotspot=15 file=2]`@RegisterProvider` annotation tells the framework to register the provider classes to be used when the framework invokes the interface. You can add as many providers as necessary.
-In the [hotspot=17-22 file=2]`SystemClient` interface, add a response exception mapper as a provider to map the `404` response code with the [hotspot=21 file=2]`UnknownUrlException` exception.
+The [hotspot=RegisterProvider file=2]`@RegisterProvider` annotation tells the framework to register the provider classes to be used when the framework invokes the interface. You can add as many providers as necessary.
```
Comment (t-koman) [inline README.adoc]: change
The [hotspot=RegisterProvider file=2]`@RegisterProvider` annotation tells the framework to register the provider classes to be used when the framework invokes the interface.
TO
**The [hotspot=RegisterProvider file=2]`@RegisterProvider` annotation indicates to the framework to register the provider classes that framework is to use when it invokes the interface.**

File: README.adoc [hunk 7]
```diff
@@ -111,23 +113,28 @@ The code for the `system` service in the `src/main/java/io/openliberty/guides/sy
...
 SystemClient.java
-[source, Java, linenums, role='code_column']
+[source, Java, linenums, role='code_column hide_tags=copyright']
 ----
-include::finish/src/main/java/io/openliberty/guides/inventory/client/SystemClient.java[tags=**;!copyright]
+include::finish/src/main/java/io/openliberty/guides/inventory/client/SystemClient.java[]
 ----
 
-The MicroProfile Rest Client feature automatically builds and generates a client implementation based on what is defined in the [hotspot=17-22 file=2]`SystemClient` interface. Thus, you don't waste time on setting up the client and connecting with the remote service.
+The MicroProfile Rest Client feature automatically builds and generates a client implementation based on what is defined in the [hotspot=SystemClient file=2]`SystemClient` interface. Thus, you don't waste time on setting up the client and connecting with the remote service.
```
Comment (t-koman) [inline README.adoc]: change
Thus, you don't waste time on setting up the client and connecting with the remote service.
TO
**There is no need to to set up the client and connect with the remote service.**

File: README.adoc [hunk 8]
```diff
@@ -178,60 +185,63 @@ The [hotspot=15-18 file=2]`handles()` method inspects the HTTP response code to
...
+The [hotspot=RegisterRestClient file=1]`@RegisterRestClient` annotation registers the interface as a RESTful client. The runtime creates a CDI managed bean for every interface that is annotated with the `@RegisterRestClient` annotation.
 
-The [hotspot=14 file=1]`@RegisterRestClient` annotation registers the interface as a RESTful client. The runtime creates a CDI managed bean for every interface that is annotated with the `@RegisterRestClient` annotation.
+The [hotspot=RegisterRestClient file=1]`configKey` value in the [hotspot=RegisterRestClient file=1]`@RegisterRestClient` annotation replaces the fully-qualified classname of the properties in the [hotspot file=0]`microprofile-config.properties` configuration file.
+For example, the `<fully-qualified classname>/mp-rest/uri` property becomes `systemClient/mp-rest/uri`.
+The benefit of using Config Keys is when multiple client interfaces have the same `configKey` value, the interfaces can be configured with a single MP config property.
 
-The [hotspot=13 file=1]`@Dependent` and [hotspot=14 file=1]`@RegisterRestClient` annotations imply that the interface is manageable through CDI. You must have them in order to inject the client.
```
[the remark this answers is not in the document]
Comment (evelinec) [reply README.adoc]: This is the deleted/replaced changes (highlight in pink).

File: README.adoc [hunk 9]
```diff
@@ -178,60 +185,63 @@ The [hotspot=15-18 file=2]`handles()` method inspects the HTTP response code to
...
 
-Inject the [hotspot=17-22 file=1]`SystemClient` interface into the [hotspot=25-102 file=2]`InventoryManager` class, which is another CDI managed bean. 
+The [hotspot=RegisterRestClient file=1]`@RegisterRestClient` annotation, which is a bean defining annotation implies that the interface is manageable through CDI. You must have it in order to inject the client.
```
[the remark this answers is not in the document]
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc [hunk 10]
```diff
@@ -178,60 +185,63 @@ The [hotspot=15-18 file=2]`handles()` method inspects the HTTP response code to
...
+include::finish/src/main/java/io/openliberty/guides/inventory/InventoryManager.java[]
 ----
 
-[hotspot=30 file=2]`@Inject` and [hotspot=31 file=2]`@RestClient` annotations inject an instance of the [hotspot=32 file=2]`SystemClient` called `defaultRestClient` to the [hotspot=25-99 file=2]`InventoryManager` class.
+[hotspot=Inject file=2]`@Inject` and [hotspot=RestClient file=2]`@RestClient` annotations inject an instance of the [hotspot=SystemClient file=2]`SystemClient` called `defaultRestClient` to the [hotspot file=2]`InventoryManager` class.
 
-Since the [hotspot=25-102 file=2]`InventoryManager` class is [hotspot=24 file=2]`@ApplicationScoped`, and the [hotspot=17-22 file=1]`SystemClient` CDI bean maintains the same scope through the [hotspot=13 file=1]`@Dependent` annotation, the client is initialized once per application.
+Since the [hotspot file=2]`InventoryManager` class is [hotspot=ApplicationScoped file=2]`@ApplicationScoped`, and the [hotspot=SystemClient file=1]`SystemClient` CDI bean maintains the same scope through the default dependent scope, the client is initialized once per application.
```
[the remark this answers is not in the document]
Comment (evelinec) [reply README.adoc]: Partially updated. The `use of` doesn't fit.

File: README.adoc (hunk 1 above)
Comment (evelinec) [reply README.adoc]: Adjusted

File: README.adoc (hunk 2 above)
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc (hunk 3 above)
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc (hunk 4 above)
Comment (evelinec) [reply README.adoc]: The original sentence is clearer.

File: README.adoc (hunk 5 above)
Comment (evelinec) [reply README.adoc]: The original is clear.

File: README.adoc (hunk 6 above)
Comment (evelinec) [reply README.adoc]: The original text is correct.

File: README.adoc (hunk 7 above)
Comment (evelinec) [reply README.adoc]: The original text is correct and fit the context.

**Comment by evelinec** [comment]
@t-koman Thank you for the feedback. Most of them had been addressed in https://github.com/OpenLiberty/guide-microprofile-rest-client/pull/102. Can you take a look pls? Thank you.

**Comment by DannyS03** [comment]
- tested and verified end-to-end on master
- LGTM.

**Pull request merged**